File: docking/hw_testing/docking.py
import asyncio

from drone import Drone
from target import Target

async def dock():
    target = Target(0, 0, 0, 0)
    drone = Drone(target)
    await drone.connect_gazebo()
    await drone.takeoff(1)
    
if __name__ == "__main__":
    loop = asyncio.get_event_loop()
    loop.run_until_complete(dock())

File: docking/cpp_docking/src/docking.cpp
#include "drone.hpp"
#include "camera_simulator.hpp" // temporary

int main(/*int argc, char** argv */)
{
    Target t;
    // t.lat = 2;
    //t.yaw = 180;

    Drone drone(t);
    drone.connect_gazebo();
    drone.arm();
    drone.takeoff();
    drone.initiate_docking(1);
    //drone.test2();

    return 0;
}